src/cpp/rtps/RTPSDomain.cpp

```diff
@@ -317,6 +344,108 @@ bool RTPSDomain::removeRTPSReader(
     return false;
 }
 
+RTPSParticipant* RTPSDomain::rosEnvironmentCreationOverride(
+        uint32_t domain_id,
+        const RTPSParticipantAttributes& att,
+        RTPSParticipantListener* listen /*= nullptr*/)
+{
+    // Check the specified discovery protocol: if other than simple it has priority over ros environment variable
+    if(att.builtin.discovery_config.discoveryProtocol != DiscoveryProtocol_t::SIMPLE)
+    {
+        logInfo(DOMAIN, "Detected non simple discovery protocol attributes."
+            << " Ignoring ros2 default client-server setup.");
+        return nullptr;
+    }
+
+    // Check for the environment variable
+    #pragma warning(suppress:4996)
+    const char* address = std::getenv(DEFAULT_ROS2_MASTER_URI);
+
+    if( nullptr == address )
+    {
+        // Back to default creation procedure
+        return nullptr;
+    }
+
+    logInfo(DOMAIN, "Detected ROS client-server environment variable. Trying to setup client-server default setup.");
+
+    // Parse the IPv4Address and port.
+    std::cmatch mr;
+    std::string ip_address;
+    int port = DEFAULT_ROS2_SERVER_PORT;
+
+    if(regex_match(address, mr, ROS2_IPV4_PATTERN))
+    {
+        std::cmatch::iterator it = mr.cbegin();
+        ip_address = (++it)->str();
+
+        if((++it)->matched)
+        {
+            port = std::stoi(it->str());
+        }
+    }
+    else
+    {
+        logError(DOMAIN, "The environment variable '" << DEFAULT_ROS2_MASTER_URI
+            << "' has an invalid IPv4 pattern '" << address << "'");
+        return nullptr;
+    }
+
+    // Check if the IP address belong to the local interfaces
+    RTPSParticipant* part = nullptr;
+    LocatorList_t locals;
+    Locator_t server_address(port);
+    IPLocator::setIPv4(server_address, ip_address);
+
+    IPFinder::getIP4Address(&locals);
+    if(locals.end() != std::find_if(locals.begin(), locals.end(),
+        [&server_address](const Locator_t & loc) -> bool
+        {
+            return IPLocator::compareAddress(server_address, loc);
```

Comment:
You'd need to compare full locators, at least on my machine
```suggestion
            return IPLocator::compareAddress(server_address, loc, true);
```

Reply:
nope, those locators have 0 port because is a list of interfaces.